src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -13071,13 +13133,16 @@ class CL0<T>
 " }, options: WithNonNullTypesTrue());
 
             c.VerifyDiagnostics(
-                // (12,16): warning CS8620: Nullability of reference types in argument of type 'CL0<string?>' doesn't match target type 'CL0<string>' for parameter 'x' in 'void C.M1(ref CL0<string> x)'.
+                // (12,16): warning CS8620: Differences in nullability of reference types in argument of type 'CL0<string?>' prevent it from being used as an input of type 'CL0<string>' for parameter 'x' in 'void C.M1(ref CL0<string> x)'.
                 //         M1(ref x1);
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInArgument, "x1").WithArguments("CL0<string?>", "CL0<string>", "x", "void C.M1(ref CL0<string> x)").WithLocation(12, 16),
-                // (19,16): warning CS8620: Nullability of reference types in argument of type 'CL0<string>' doesn't match target type 'CL0<string?>' for parameter 'x' in 'void C.M2(out CL0<string?> x)'.
+                // (12,16): warning CS8624: Differences in nullability of reference types in argument of type 'CL0<string?>' prevent it from being used as an output of type 'CL0<string>' for parameter 'x' in 'void C.M1(ref CL0<string> x)'.
```

Comment:
>output [](start = 158, length = 6)

Having the one difference between these error messages be 19 words in makes these appear to be duplicates at first blush. Perhaps reword to be `'CL0<string?>' cannot be used an input/output type of 'CL0<string>' for parameter 'x' in 'void C.M1(ref CL0<string> x)' due to differences in the nullability of reference types.` This moves the different word to 6 words in. Not perfect, but I can't think of a better wording off the top of my head that puts it first or second.

Reply:
Or, after having looked at `SuppressNullableWarning_Ref_WithNestedDifferences`, perhaps a wording that puts the method name at the end, but still keeps `input/output` as close to the front of the error as possible.

---
In reply to: [258173669](https://github.com/dotnet/roslyn/pull/33447#discussion_r258173669) [](ancestors = 258173669)

Reply:
I like it! Thanks for the concrete proposal. 

---
In reply to: [258193243](https://github.com/dotnet/roslyn/pull/33447#discussion_r258193243) [](ancestors = 258193243,258173669)